typescript/mcap/src/v0/Mcap0IndexedReader.ts

```diff
@@ -1,11 +1,216 @@
 import { crc32, crc32Final, crc32Init, crc32Update } from "@foxglove/crc";
 import Heap from "heap-js";
+import { sortedIndexBy } from "lodash";
 
-import { getBigUint64 } from "../common/getBigUint64";
-import { MCAP0_MAGIC, Opcode } from "./constants";
+import { MCAP0_MAGIC } from "./constants";
 import { parseMagic, parseRecord } from "./parse";
 import { DecompressHandlers, IReadable, TypedMcapRecords } from "./types";
 
+type ChunkCursorParams = {
+  chunkIndex: TypedMcapRecords["ChunkIndex"];
+  relevantChannels: Set<number> | undefined;
+  startTime: bigint | undefined;
+  endTime: bigint | undefined;
+};
+
+class ChunkCursor {
```

Comment:
Please move this to a separate file. I am a fan of keeping files smaller by keeping 1 "main" class per file. This class has grown sizable enough that it is easier to maintain as a standalone file.

Reply:
A comment for this class would be useful to future maintainers.

Reply:
yea, was planning to do that, this PR is still a draft